kpi/tests/api/v2/test_api_submissions.py

```diff
@@ -987,6 +990,38 @@ def test_get_edit_link_response_includes_csrf_cookie(self):
         assert ENKETO_CSRF_COOKIE_NAME in response.cookies
         assert response.cookies[ENKETO_CSRF_COOKIE_NAME].value
 
+    def test_edit_submission_with_wrong_digest_credentials(self):
```

Comment:
as we discussed, this is a good test that handles 3 authentication scenarios. but there are more! :scream_cat: 

Session | Digest |  
-- | -- | --
unauth | unauth | done
good | unauth |  
no perm | unauth |  
unauth | good | done
good | good |  
no perm | good |  
unauth | no perm | done
good | no perm |  
no perm | no perm |  